src/meta/cluster_balance_policy.cpp

```diff
@@ -544,10 +545,21 @@ bool cluster_balance_policy::apply_move(const move_info &move,
     // add into migration list and selected_pid
     partition_configuration pc;
     pc.pid = move.pid;
-    pc.primary = primary_addr;
-    list[move.pid] = generate_balancer_request(*_global_view->apps, pc, move.type, source, target);
+    pc.hp_primary = primary_hp;
+    std::shared_ptr<dsn::dns_resolver> resolver;
+    if (_svc == nullptr) {
+        resolver = std::make_shared<dsn::dns_resolver>();
```

Comment:
In which case the `_svc` is nullptr?

Reply:
This is a defensive coding, and there are cases in the test where `_svc` is nullptr.

Reply:
How about creating a valid meta_service there and pass it to this function, to ensure consistence.